spec/index.bs

```diff
@@ -1066,29 +1074,26 @@ a better limit is not specified.
         when creating a {{GPURenderPipeline}}.
 </table>
 
-#### <dfn dictionary>GPULimits</dfn> #### {#gpulimits}
+#### <dfn interface>GPUAdapterLimits</dfn> #### {#gpu-adapterlimits}
 
-{{GPULimits}} describes the [=limits=] with which a device should be created.
-See {{GPUDeviceDescriptor/limits|GPUDeviceDescriptor.limits}}.
-
-It also exposes the [=limits=] supported by an adapter.
+{{GPUAdapterLimits}} exposes the [=limits=] supported by an adapter.
 See {{GPUAdapter/limits|GPUAdapter.limits}}.
 
 <script type=idl>
-dictionary GPULimits {
-    GPUSize32 maxBindGroups = 4;
-    GPUSize32 maxDynamicUniformBuffersPerPipelineLayout = 8;
-    GPUSize32 maxDynamicStorageBuffersPerPipelineLayout = 4;
-    GPUSize32 maxSampledTexturesPerShaderStage = 16;
-    GPUSize32 maxSamplersPerShaderStage = 16;
-    GPUSize32 maxStorageBuffersPerShaderStage = 4;
-    GPUSize32 maxStorageTexturesPerShaderStage = 4;
-    GPUSize32 maxUniformBuffersPerShaderStage = 12;
-    GPUSize32 maxUniformBufferBindingSize = 16384;
-    GPUSize32 maxStorageBufferBindingSize = 134217728;
-    GPUSize32 maxVertexBuffers = 8;
-    GPUSize32 maxVertexAttributes = 16;
-    GPUSize32 maxVertexBufferArrayStride = 2048;
+interface GPUAdapterLimits {
```

Comment:
why does this need to be an interface?

Reply:
Perhaps it doesn't, but dictionaries are clearly designed as an input type and not an output type in IDL. One problem with a dictionary is that `adapter.limits` can't be `[SameObject]`, which I think means returning a new object every time, which would be surprising e.g. if an app tries to hang extra properties off of it.

Also I believe it also means `adapter.limits` must return a vanilla plain-old-data object, according to the [IDL-to-JS conversion](https://heycam.github.io/webidl/#es-dictionary), which doesn't leave it open for instrumentation. Arguably we could return an `object` with attributes attached, but at that point it's basically identical to an `interface`.

Reply:
That makes sense